asteroid_filterbanks/pcen.py

```diff
@@ -0,0 +1,162 @@
+            self.weights = nn.Parameter(
+                torch.full((n_channels,), fill_value=smooth), requires_grad=trainable
+            )
+        else:
+            self.weights = nn.Parameter(
+                torch.full((1,), fill_value=smooth), requires_grad=trainable
+            )
+
+    def forward(self, mag_spec, initial_state):
+        weights = torch.clamp(self.weights, 0.0, 1.0)
+        accumulator = initial_state
+        out = None
+        for x in torch.split(mag_spec, 1, dim=-1):
+            accumulator = weights * x + (1.0 - weights) * accumulator
+            if out is None:
+                out = accumulator
+            else:
+                out = torch.cat((out, accumulator), dim=-1)
+        return out
+
+
+class TrainableParameters(TypedDict):
+    alpha: bool
+    delta: bool
+    root: bool
+    smooth: bool
+
+
+class PCEN(nn.Module):
+    """Per-Channel Energy Normalization as described in [1].
+
+    This applies a fixed or learnable normalization by an exponential moving average smoother, and a compression.
+
+    Args:
+        alpha: The AGC strength (or gain normalization strength) (α in the paper).
+            Defaults to 0.96
+        delta: Bias added before compression (δ in the paper).
+            Defaults to 2.0
+        root: One over exponent applied for compression (r in the paper).
+            Defaults to 2.0
+        floor: Offset added to compression, prevents division by zero. (ϵ in the paper)
+            Defaults to 1e-6
+        smooth: Smoothing coefficient (s in the paper).
+            Defaults to 0.04
+        n_channels: Number of channels in the time frequency representation.
+            Defaults to 2
+        trainable: If True, the parameters (alpha, delta, root and smooth) are trainable. If False, the parameters are fixed.
+            Individual parameters can set to be fixed or trainable by passing a dictionary of booleans, with the key
+            matching the parameter name and the value being either True (trainable) or False (fixed).
+            i.e. ``{"alpha": False, "delta": True, "root": False, "smooth": True}``
+            Defaults to False
+        per_channel_smoothing: If True, means each channel has it's own smooth coefficient.
+            Defaults to False
+
+    References
+        [1]: Wang, Y., et al. "Trainable Frontend For Robust and Far-Field Keyword Spotting”, arXiv e-prints, 2016.
+             https://arxiv.org/pdf/1607.05666.pdf
+
+    Influenced by leaf-audio's tensorflow implementation
+        https://github.com/google-research/leaf-audio/blob/7ead2f9fe65da14c693c566fe8259ccaaf14129d/leaf_audio/postprocessing.py
+
+    See license here
+        https://github.com/google-research/leaf-audio/blob/master/LICENSE
+    """
+
+    def __init__(
+        self,
+        alpha: float = 0.96,
+        delta: float = 2.0,
+        root: float = 2.0,
+        floor: float = 1e-6,
+        smooth: float = 0.04,
+        n_channels: int = 2,
```

Comment:
Why is 2 the default?

Reply:
No particular reason actually! Do you think another value is a more sensible default? Maybe `1`?

Reply:
I'd do `1` yes, please